connectors/src/api/slack_channels_linked_with_agent.ts

```diff
@@ -66,19 +68,39 @@ const _patchSlackChannelsLinkedWithAgentHandler = async (
     new Set(slackChannelIds.filter((id) => !foundSlackChannelIds.has(id)))
   );
 
-  if (missingSlackChannelIds.length) {
-    return apiError(req, res, {
-      api_error: {
-        type: "not_found",
-        message: `Slack channel(s) not found: ${missingSlackChannelIds.join(
-          ", "
-        )}`,
-      },
-      status_code: 404,
-    });
-  }
-
   await sequelize_conn.transaction(async (t) => {
+    if (missingSlackChannelIds.length) {
+      const remoteChannels = (
+        await getChannels(parseInt(connectorId), false)
+      ).flatMap((c) => (c.id && c.name ? [{ id: c.id, name: c.name }] : []));
+      const remoteChannelsById = remoteChannels.reduce((acc, ch) => {
+        acc[ch.id] = ch;
+        return acc;
+      }, {} as Record<string, { id: string; name: string }>);
+      const createdChannels = await Promise.all(
+        missingSlackChannelIds.map((slackChannelId) => {
+          const remoteChannel = remoteChannelsById[slackChannelId];
+          if (!remoteChannel) {
+            throw new Error(
+              `Unexpected error: Access to the Slack channel ${slackChannelId} seems lost.`
```

Comment:
If we thow no channel will be attached to the agent and the error is managed from Front?